Engine_Revit_UI/Query/ElementIds/ElementIdsByView.cs

```diff
@@ -43,10 +45,17 @@ public static partial class Query
         /****              Public methods               ****/
         /***************************************************/
 
-        public static IEnumerable<ElementId> ElementIdsViewTemplate(this Document document, IEnumerable<ElementId> ids = null)
+        [Description("Get all Elements as ElementId that are visible in View")]
+        [Input("document", "Revit Document where views are collected")]
+        [Output("elementIdsByView", "An enumerator for easy iteration of ElementIds collected")]
+        public static IEnumerable<ElementId> ElementIdsByView(this Document document, View view)
```

Comment:
Agreed changed the name. I implemented ids == null, but in a not very elegant way because `FilteredElementCollector` uses document and View.Id in this case, so I thought that the IEnumerable ids could instead be used to narrow further, e.g. look for these ElementIds that are visible in this View. What do you think?
```
FilteredElementCollector collector = new FilteredElementCollector(document, view.Id);
 if (ids == null)
{
                return collector.ToElementIds();
}
else
{
                collector.Where(x => ids.Contains(x.Id)).ToList();
                return collector.ToElementIds();
}
```

Reply:
HashSet can has a nice [`IntersectWith`](https://docs.microsoft.com/pl-pl/dotnet/api/system.collections.generic.hashset-1.intersectwith?view=netframework-4.8) method, I used in a few places - might be what you are looking for?

Reply:
Yes indeed, I've implemented it!